docs/application/dotnet/guides/multimedia/webrtc.md

```diff
@@ -0,0 +1,509 @@
+
+    ```csharp
+    mediaSource.VideoResolution = new Size(640, 480);
+    ```
+    > [!NOTE]
+    > Some types of media sources support dynamic resolution change while streaming. Otherwise `InvalidOperationException` exception will be thrown.
+
+8. To set or get the video frame rate to a media source, use the `VideoFrameRate()` property:
+
+    ```csharp
+    mediaSource.VideoFrameRate = 15;
+    ```
+    > [!NOTE]
+    > If the input value is not supported by the media source, the `ErrorOccurred` event will be invoked.
+
+9. To set the device id to a media source for camera, use the `CameraDeviceId` property before calling `Start()` or `StartAsync()`:
+
+    ```csharp
+    ...
+    var cameraSource = new MediaCameraSource();
+    webRtc.AddSource(cameraSource);
+    ...
+    cameraSource.CameraDeviceId = 1;
+    ...
+    await webRtc.StartAsync();
+    ```
+    > [!NOTE]
+    > If the device id is not valid, `StartAsync()` or `Start` will throw `InvalidOperationException` exception.
+
+<a name="data_channel"></a>
+## Control data channels
+
+You can create a data channel to a webrtc. It is also possible to get notified when you have a new data channel requested by a remote peer. You can send or receive data to or from these data channels by using the APIs below:
+
+1. To create a data channel, use `WebRTCDataChannel` before calling `Start()` or `StartAsync()`:
+
+    ```csharp
+    var webRtc = new WebRTC();
+    var dataChannel = new WebRTCDataChannel(webRtc, "data_channel_label");
+    ...
+    await webRtc.StartAsync();
+    ```
+
+2. To get notified when a data channel is created by a remote peer, use `WebRTC.DataChannel` event. The event will be invoked when it is created after negotiation:
+
+    ```csharp
+    webRtc.DataChannel += (s, e) => Log.Info("WebRTC", "DataChannel is created");
+
+    await webRtc.StartAsync();
+    ```
+
+3. Once you get a data channel as above, you can handle the following events:
+
+    ```csharp
+    webRtc.DataChannel += (s, e) =>
+    {
+        e.DataChannel.Opened += (s, e) => Log.Info("WebRTC", "Opened");
+        e.DataChannel.Closed += (s, e) => Log.Info("WebRTC", "Closed");
+        e.DataChannel.ErrorOccurred += (s, e) => Log.Info("WebRTC", $"{e.Error}");
+        e.DataChannel.MessageReceived += (s, e) =>
+        {
+            switch(e.Type)
+            {
+                case DataChannelType.Strings:
+                    Log.Info("WebRTC", $"{e.Message}");
+                    break;
+                case DataChannelType.Bytes:
+                    var data = e.Data;
+                    // do something
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    await webRtc.StartAsync();
+    ```
+
+4. To send string data to the data channel, use `Send()`:
+
+    ```csharp
+    dataChannel.Send("string_to_send");
+    ```
+
+5. To send bytes data to the data channel, use `webrtc_data_channel_send_bytes()`:
```

Comment:
```suggestion
5. To send byte data to the data channel, use `webrtc_data_channel_send_bytes()`:
```

Reply:
Hi @hsgwon , I have another question here. The function `webrtc_data_channel_send_bytes()` declared here is not being used in the code below. Is this okay?

Reply:
I'll change it to `Send()`